pkg/diagnostics/networkpod/pod.go

```diff
@@ -0,0 +1,288 @@
+	}
+}
+
+func checkSameNodePodToPodConnection(globalPods, nonGlobalPods []kapi.Pod, vmap map[string]uint32, r types.DiagnosticResult) {
+	// Applicable to flat and multitenant networks
+	if len(nonGlobalPods) >= 2 {
+		checkPodToPodConnection(nonGlobalPods[0], nonGlobalPods[1], vmap, r)
+	} else {
+		r.Warn("DPodNet1008", nil, fmt.Sprintf("Skipping pod connectivity test for non-global projects on the same node. Reason: Couldn't find 2 non-global pods."))
+	}
+	if len(vmap) == 0 {
+		return
+	}
+
+	// Applicable to multitenant network
+	if len(globalPods) >= 2 {
+		checkPodToPodConnection(globalPods[0], globalPods[1], vmap, r)
+	} else {
+		r.Warn("DPodNet1009", nil, fmt.Sprintf("Skipping pod connectivity test for global projects on the same node. Reason: Couldn't find 2 global pods."))
+	}
+
+	if len(globalPods) > 0 && len(nonGlobalPods) > 0 {
+		checkPodToPodConnection(globalPods[0], nonGlobalPods[0], vmap, r)
+	} else {
+		r.Warn("DPodNet1010", nil, fmt.Sprintf("Skipping pod connectivity test between global to non-global projects on the same node. Reason: Couldn't find global and/or non-global pod."))
+	}
+}
+
+// Determine expected connection status for the given pods
+// true indicates success and false means failure
+func expConnStatus(ns1, ns2 string, vmap map[string]uint32) bool {
+	// Check if sdn is flat network
+	if len(vmap) == 0 {
+		return true
+	} // else multitenant
+
+	// Check if one of the pods belongs to global network
+	if vmap[ns1] == api.GlobalVNID || vmap[ns2] == api.GlobalVNID {
+		return true
+	}
+
+	// Check if both the pods are sharing the network
+	if vmap[ns1] == vmap[ns2] {
+		return true
+	}
+
+	// Isolated network
+	return false
+}
+
+func printPod(pod kapi.Pod) string {
+	return fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
+}
+
+// checkPodToPodConnection verifies connection from fromPod to toPod.
+// Connection check from toPod to fromPod will be done by the node of toPod.
+func checkPodToPodConnection(fromPod, toPod kapi.Pod, vmap map[string]uint32, r types.DiagnosticResult) {
+	if len(fromPod.Status.ContainerStatuses) <= 0 {
+		r.Error("DPodNet1011", nil, fmt.Sprintf("ContainerID not found for pod %q", printPod(fromPod)))
+		return
+	}
+
+	success := expConnStatus(fromPod.Namespace, toPod.Namespace, vmap)
+
+	kexecer := kexec.New()
+	containerID := kcontainer.ParseContainerID(fromPod.Status.ContainerStatuses[0].ContainerID).ID
+	pid, err := kexecer.Command("docker", "inspect", "-f", "{{.State.Pid}}", containerID).CombinedOutput()
+	if err != nil {
+		r.Error("DPodNet1012", err, fmt.Sprintf("Fetching pid for pod %q, container %q failed. Error: %s", printPod(fromPod), containerID, err))
+		return
+	}
+
+	out, err := kexecer.Command("nsenter", "-n", "-t", strings.Trim(fmt.Sprintf("%s", pid), "\n"), "--", "ping", "-c1", "-W1", toPod.Status.PodIP).CombinedOutput()
```

Comment:
Okay, launching our own pods also solves the problem of not finding global/non-global pods which will skip some of the tests.
May be I could do:
 - Create 2 namespaces for network diagnostics: one with global access and other one isolated.
 - Launch one pod and one service in diags global namespace and isolated namespace on every node. This will ensure we run all network diagnostics tests on all nodes.